server/sockets.js

```diff
@@ -0,0 +1,39 @@
+const socketIo = require("socket.io");
+
+class SocketConfig {
+  constructor() {
+    this.io;
+  }
+
+  bindServer(server) {
+    this.io = new socketIo(server);
+  }
+
+  start() {
+    this.io.on("connection", (socket) => {
+      socketListeners(socket);
+    });
+  }
+
+  emit(event, data) {
+    this.io.emit(event, data);
+  }
+
+  sendNotification(room, data) {
+    this.io.to(room).emit("notification", data);
+  }
+}
+
+const socketListeners = (socket) => {
+  console.log("User connected via sockets");
+
+  // Subscribe after login:
+  socket.on("login", (userId) => {
+    console.log("User joined room " + userId);
+    socket.join(userId);
```

Comment:
Ideally you would check the JWT token in the cookies sent via sockets, but this is something I would table to work on later given the time frame we have left

Reply:
removed for now